sql/threadpool_unix.cc

```diff
@@ -950,6 +946,16 @@ static void thread_group_close(thread_group_t *thread_group) noexcept {
 
   mysql_mutex_unlock(&thread_group->mutex);
 
+  mysql_mutex_lock(&thread_group->mutex);
```

Comment:
Locking `thread_group->mutex` here just after unlocking it. May be remove both calls?

Reply:
On purpose, this is the 1st chance for those worker threads to enter their pre-exit critical sections. I can remove it if you still want that